Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py

```diff
@@ -147,6 +149,36 @@ def git(cls, args, repository, additional_setup=None, hooks=None, **kwargs):
             sys.stderr.write('Failed to use {} as the merge strategy\n'.format('merge commits' if args.merge else 'rebase'))
             result += 1
 
+        if args.defaults:
+            pr_technique = repository.config(location='project')['webkitscmpy.pull-request']
+        else:
+            pr_technique = Terminal.choose(
+                'When editing pull-requests, would you like to overwrite existing commits or append to the branch?',
+                default=repository.config(location='project')['webkitscmpy.pull-request'],
+                options=['overwrite', 'append'],
+            )
+        if run(
+            [local.Git.executable(), 'config', 'webkitscmpy.pull-request', pr_technique],
+            capture_output=True, cwd=repository.root_path,
+        ).returncode:
+            sys.stderr.write("Failed to set '{}' as the default pull request technique\n".format(pr_technique))
+            result += 1
+
+        if args.defaults or pr_technique == 'append' or repository.config(location='project')['webkitscmpy.history'] == 'never':
+            pr_history = repository.config(location='project')['webkitscmpy.history']
+        else:
+            pr_history = Terminal.choose(
+                'Would you like to retain history when you overwrite a pull request branch?',
+                default=repository.config(location='project')['webkitscmpy.history'],
+                options=repository.PROJECT_CONFIG_OPTIONS['webkitscmpy.history'],
+            )
```

Comment:
What's the intended use case here? I'm not sure I understand why a dev would want to amend a commit _but also_ push a historical record elsewhere. If they want to preserve history, then just push a new commit ;)

If the goal is to parallel the "Show Obsolete" view in bugzilla, could we direct folks to the commit saved by GitHub when it detects a force push?
![Screen Shot 2022-01-24 at 15 42 46](https://user-images.githubusercontent.com/96081375/150883786-09cc2938-5615-42cb-9f90-4026123ab691.png)

Reply:
This is basically a carve-out for folks who want to use the force-push approach. It's quite a bit more persistent than the commit saved by GitHub when it detects a force push, I've noticed those aren't accessible in certain circumstances (and I'm not exactly sure which circumstances those are). I know this is a feature I've used in the last few weeks as I've been iterating on changes, especially when the feedback is "this needs to be done a totally different way," and "a totally different way" was something I had tried in an earlier PR draft.